omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierResource.java

```diff
@@ -10,16 +10,21 @@
 package org.openmrs.module.idgen.rest.resource;
 
 import org.openmrs.api.context.Context;
+import org.openmrs.module.idgen.Identifier;
+import org.openmrs.module.idgen.IdentifierSource;
 import org.openmrs.module.idgen.service.IdentifierSourceService;
 import org.openmrs.module.webservices.rest.SimpleObject;
 import org.openmrs.module.webservices.rest.web.RequestContext;
-import org.openmrs.module.webservices.rest.web.resource.api.SubResource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource;
 import org.openmrs.module.webservices.rest.web.response.ResponseException;
 
 @org.openmrs.module.webservices.rest.web.annotation.SubResource(parent = IdentifierSourceResource.class, path =
-        "identifier", supportedClass = String.class, supportedOpenmrsVersions = {"1.8.*", "1.9.*", "1.10.*", "1.11.*", 
+        "identifier", supportedClass = Identifier.class, supportedOpenmrsVersions = {"1.8.*", "1.9.*", "1.10.*", "1.11.*", 
         		"1.12.*", "2.0.*", "2.1.*", "2.2.*" })
-public class IdentifierResource implements SubResource {
+public class IdentifierResource extends DelegatingSubResource<Identifier, IdentifierSource, IdentifierSourceResource> {
```

Comment:
@dkayiwa  if we are to add the IdentifierResource to the docs, we need to extend the BaseDelegatingResource inorder to override the required methods, that being said the IdentifierResource was supporting the String class(identifier string) incase we extend the BaseDelegatingResource class with the String class as the type we are delegating to, we shall get duplicate methods here https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/BaseDelegatingResource.java#L544-L566

Reply:
Hence hitting compiling errors